src/main.rs

```diff
@@ -26,20 +32,31 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     collector.init();
 
-    let broker = IntentBroker::new();
-    let registry = Registry::new(broker.clone());
+    let ess = SharedEss::new();
+    let broker = IntentBroker::new(
+        format!(
+            "http://{}:{}",
+            env(EXTERNAL_HOST_NAME_ENV).unwrap_or_else(|| "localhost".to_string()),
```

Comment:
You can avoid allocating the `"localhost"` string given that it's a literal:

```suggestion
            env::<String>(EXTERNAL_HOST_NAME_ENV).as_deref().unwrap_or("localhost"),
```